### PR TITLE
rpc: support (but don't require) simplejson

### PR DESCRIPTION
**Pull request by welshjf**
Switching to `simplejson` seems to improve CPU and memory use, particularly for the expensive `getrawmempool(True)`.

I also tried `ujson` but it doesn't support the `parse_float` parameter.

### CODE REVIEW

File: bitcoin/rpc.py [hunk 1]
```diff
@@ -10,7 +10,8 @@
 # propagated, or distributed except according to the terms contained in the
 # LICENSE file.
 
-"""Bitcoin Core RPC support"""
+"""Bitcoin Core RPC support. By default this uses the standard library ``json``
```
Comment (petertodd) [inline bitcoin/rpc.py]: Make "By default" on a separate line, with a space between it and the title  "Bitcoin Core RPC support"

**Comment by petertodd** [comment]
Seems like a reasonable idea; simplejson is supported on both python3 and python2.

It'd be better though if this could be an opt-in - any ideas on making that possible? If there's a json bug people could of course lose money here...

**Comment by welshjf** [comment]
Opt-in was indeed my intention, but on a fresh look, my approach is perhaps overly magical. Explicit is better than implicit, as they say in Python. If the user wants to use two different json libs at once, it's not for us to say "no".

The difficulty is, that it's nice to have everything imported early at global scope, such as for ease of reference, and static checkers.

Perhaps initially always import json, then have a function to replace it with something else. An added benefit is this will support any (API-compatible) json lib, at the user's risk.

**Comment by welshjf** [comment]
In fact, this is already possible with no new code:

    import bitcoin.rpc
    import simplejson
    bitcoin.rpc.json = simplejson

Can't say I exactly like it, but haven't got anything better. Does this seem reasonable to document as a supported thing?

**Comment by petertodd** [comment]
Sure, let's do that.

**Comment by welshjf** [comment]
Updated.

**Pull request merged**